### DISCUSSION TITLE
Can't create a filament user

### DISCUSSION THREAD

**Discussion by robindelaater** [filamentphp/filament#10369] [Help] [answered]
Hello! 
I'm having trouble creating a user after deploying my project.
Locally I've got no issues at all, installed filament and can run the `artisan make:filament-user` command.

But after deploying I'm seeing the following in red: `Filament has not been installed yet: php artisan filament:install --panels`.
I already have filament installed, and also tried running the suggested command without luck.

Any suggestions?

P.S.
Let me know if any more info is required, this is my first time doing something like this.

**Comment by leandrocfe** [comment]
`php artisan about`
What is the output? Is there filament installed? What is the version?
Did you also run `composer install`?

**Reply by robindelaater** [reply]
Hey @leandrocfe,

Thanks for getting back to me, see the output of php artisan about below:

`Environment ......................................................................................................................................  
  Application Name ......................................................................................................................... Laravel  
  Laravel Version .......................................................................................................................... 10.37.3  
  PHP Version ................................................................................................................................ 8.1.5  
  Composer Version ........................................................................................................................... 2.6.6  
  Environment ................................................................................................................................ local  
  Debug Mode ............................................................................................................................... ENABLED  
  URL ................................................................................................................ paint-productions.monkeydo.it  
  Maintenance Mode ............................................................................................................................. OFF  

  Cache ............................................................................................................................................  
  Config ................................................................................................................................ NOT CACHED  
  Events ................................................................................................................................ NOT CACHED  
  Routes ................................................................................................................................ NOT CACHED  
  Views ..................................................................................................................................... CACHED  

  Drivers ..........................................................................................................................................  
  Broadcasting ................................................................................................................................. log  
  Cache ....................................................................................................................................... file  
  Database ................................................................................................................................... mysql  
  Logs .............................................................................................................................. stack / single  
  Mail ........................................................................................................................................ smtp  
  Queue ................................................................................................................................... database  
  Session ..................................................................................................................................... file  

  Filament .........................................................................................................................................  
  Packages ......................................................................................... filament, forms, notifications, support, tables  
  Version .................................................................................................................................. v3.1.23  
  Views .............................................................................................................................. NOT PUBLISHED  

  Livewire .........................................................................................................................................  
  Livewire .................................................................................................................................. v3.3.0 `

I did run composer install, but that did not help.

I managed to fix it by removing everything from the server and re-deploying, so I guess we can close this topic for now.

**Comment by drbyte** [comment, marked as answer]
> `php artisan make:filament-user`
> `Filament has not been installed yet: php artisan filament:install --panels`

This will occur if you have no panels with `->default()` set.
This is included when Filament creates the first panel for your app. But if you removed that, or if you created your own panel without the artisan command, then you will encounter that error when creating a user from the artisan command

**Reply by Osub** [reply]
me too

**Reply by scottw-finao** [reply]
um, ok - and how do you fix it??? As in no ->default() set where exactly?

**Reply by drbyte** [reply]
You set it in your Panel Provider. Example in the docs:
https://filamentphp.com/docs/3.x/panels/configuration#the-default-admin-panel

**Reply by mankowitz** [reply]
@drbyte - that page has no reference to ->default()

**Reply by drbyte** [reply]
Correct. But it does explain about the default panel.
And when you create a new filament project it creates a default panel which does have `->default()` set.

But here's the stub that Filament uses to create that default/first panel upon installation:
https://github.com/filamentphp/filament/blob/3.x/packages/support/stubs/DefaultPanelProvider.stub

Quoting a section from it:
```php
class {{ class }} extends PanelProvider
{
    public function panel(Panel $panel): Panel
    {
        return $panel
            ->default()
            ->id('{{ id }}')
            ->path('{{ id }}')
            ->login()
```

**Reply by mankowitz** [reply]
That's odd. I did a clean install new laravel 10 project, and it didn't put the `->default()` line in. After I put it in, it seems to be working as expected. Thanks for the replya.

**Reply by NAPSTERSIHOTANG** [reply]
in which file you put ->default()

**Reply by drbyte** [reply]
> in which file you put ->default()

In your Filament Panel Provider. 
By default that's in `/app/Providers/Filament/`. You can see new files easily in git when you create a new Panel or any new asset.

**Comment by mgolshan** [comment]
Hi @drbyte,
I'm having the same problem.
I see that the provider has `->default()`. And, it's registered. But the make user command is failing with the same error.
```
class PanelPanelProvider extends PanelProvider
{
    public function panel(Panel $panel): Panel
    {
        return $panel
            ->default()
            ->id('panel')
            ->path('/panel')
            ->login()
            ->colors([
                'primary' => Color::Amber,
            ])
```

I also have nova installed and using a different path. I wonder if that could be related.

```
$ php artisan about

  Environment ................................................................................  
  Application Name ................................................................... My APP  
  Laravel Version .................................................................... 11.27.2  
  PHP Version .......................................................................... 8.3.4  
  Composer Version ..................................................................... 2.7.7  
  Environment .......................................................................... local  
  Debug Mode ......................................................................... ENABLED  
  URL .......................................................................... app.localhost  
  Maintenance Mode ....................................................................... OFF  
  Timezone ............................................................................... UTC  
  Locale .................................................................................. en  

  Cache .......................................................................................  
  Config .......................................................................... NOT CACHED  
  Events ........................................................................... NOT CACHED  
  Routes ........................................................................... NOT CACHED  
  Views ................................................................................ CACHED  

  Drivers .....................................................................................  
  Broadcasting ............................................................................ log  
  Cache ................................................................................. redis  
  Database ............................................................................ central  
  Logs ......................................................................... stack / single  
  Mail ................................................................................... smtp  
  Queue ................................................................................. redis
  Session ................................................................................ redis

  Filament ....................................................................................
  Packages ................................... filament, forms, notifications, support, tables  
  Version ............................................................................ v3.2.117
  Views .......................................................................... NOT PUBLISHED

  Livewire ....................................................................................
  Livewire ............................................................................. v3.5.6

  Spatie Permissions ..........................................................................
  Features Enabled .................................................................... Default
  Version ................................................................................ 6.9.0
```

**Reply by drbyte** [reply]
The error occurs when Filament looks at all the registered Panel Providers and doesn't find one that is set as default (or is not set as current based on user URI path, but that's irrelevant in command-line mode).

Is it possible in your case that your app hasn't registered all the PanelProviders yet in this command-line context? Something missing or in wrong order in your App Service Providers?

**Reply by mgolshan** [reply]
Thank you!
I noticed the file `bootstrap/providers.php` is not being used at all. I added the provider to `config/app.php` and it works now
